internal/scan-manager/scanners/common/src/main/java/org/wso2/security/tools/scanmanager/scanners/common/ScannerController.java

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc., WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ *
+ */
+
+package org.wso2.security.tools.scanmanager.scanners.common;
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.wso2.security.tools.scanmanager.scanners.common.exception.ScannerException;
+import org.wso2.security.tools.scanmanager.common.ErrorMessage;
+import org.wso2.security.tools.scanmanager.scanners.common.service.Scanner;
+import org.wso2.security.tools.scanmanager.scanners.common.util.CallbackUtil;
+import org.wso2.security.tools.scanmanager.common.ScanRequest;
+
+import java.io.IOException;
+
+/**
+ * The class {@code ScannerController} is the web controller which defines the routines for initiating
+ * scanner operations.
+ */
+@Controller
+@RequestMapping("scanner")
+public class ScannerController {
+
+    // This represents if a scan is started
+    private boolean hasScanStarted = false;
+    private static final Logger log = Logger.getLogger(ScannerController.class);
+
+    @Autowired
+    Scanner scanner;
```

Comment:
```suggestion
  public ScannerController(Scanner scanner) {
  this.scanner = scanner;
  }
```
Field injection is not recommended. Better to use the constructor injection.